src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java

```diff
@@ -0,0 +1,245 @@
+            safelyAddProperty(vendorProperties, CUPurapConstants.ZIPCODE, vendorAddress.getVendorZipCode());
+            safelyAddProperty(vendorProperties, CUPurapConstants.COUNTRY, vendorAddress.getVendorCountryCode());
+            safelyAddProperty(vendorProperties, CUPurapConstants.ADDRESS_TYPE_CODE, vendorAddress.getVendorAddressTypeCode());
+        }
+    }
+
+    private VendorAddress getVendorAddress(List<VendorAddress> vendorAddresses) {
+        for (VendorAddress address : vendorAddresses) {
+            if (address.getVendorAddressType().getVendorAddressTypeCode().equalsIgnoreCase(VendorConstants.AddressTypes.REMIT)) {
+                return address;
+            }
+        }
+        return vendorAddresses.get(0);
+    }
+
+    private void addPoItemsToProperties(Properties poProperties, List poItems) {
+        List<Properties> poLines = new ArrayList<>();
+        for (Object obj : poItems) {
+            if (obj instanceof PurApItem) {
+                PurApItem purApItem = (PurApItem)obj;
+                Properties lineProps = new Properties();
+                safelyAddProperty(lineProps, CUPurapConstants.AMOUNT.toLowerCase(), purApItem.getTotalAmount().toString());
+                safelyAddProperty(lineProps, CUPurapConstants.QUANTITY, purApItem.getItemQuantity().toString());
+                safelyAddProperty(lineProps, CUPurapConstants.UNIT_OF_MEASURE, purApItem.getItemUnitOfMeasureCode());
+                safelyAddProperty(lineProps, CUPurapConstants.DESCRIPTION, purApItem.getItemDescription());
+                lineProps.put(CUPurapConstants.UNIT_PRICE, purApItem.getItemUnitPrice());
+                lineProps.put(CUPurapConstants.LINE_NUMBER, purApItem.getItemLineNumber());
+                lineProps.put(CUPurapConstants.PART_NUMBER, purApItem.getItemAuxiliaryPartIdentifier());
+                poLines.add(lineProps);
+            }
+        }
+        poProperties.put(CUPurapConstants.ITEMS, poLines.toArray());
+    }
+
+    private String getSimpleJsonObject(String key, String value) {
+        Properties obj = new Properties();
+        obj.put(key, value);
+        return gson.toJson(obj);
+    }
+
+    private LookupDao getLookupDao() {
+        if (lookupDao == null) {
+            lookupDao = SpringContext.getBean(LookupDao.class);
+        }
+        return lookupDao;
+    }
+
+    private HashMap<String,String> parseVendorNumber(String vendorNumber) throws BadRequestException {
+        HashMap<String, String> map = new HashMap<>();
+        if (StringUtils.isEmpty(vendorNumber) || !vendorNumber.contains(CUPurapConstants.DASH)) {
+            throw new BadRequestException();
+        }
+
+        String generatedVendorHeaderId = vendorNumber.substring(0, vendorNumber.indexOf(CUPurapConstants.DASH));
+        String vendorDetailNumber = vendorNumber.substring(vendorNumber.indexOf(CUPurapConstants.DASH) + 1);
```

Comment:
StringUtils has some helper methods that can simplify getting the appropriate substrings, such as "substringBefore" and "substringAfter". Also, you may want to check that there is exactly one dash and not more than one.